janus-idp/backstage-showcase
# feat: test patching additional scaffolder permissions

## Description

This Pull Requests tests applying the additional scaffolder permissions package changes as a patch using `yarn patch-package` to test [RHIDP-2324](https://issues.redhat.com/browse/RHIDP-2324)

## Testing Instructions

The new permissions add the following new behaviours:
- Verify that new permissions are available via `<BACKEND_URL>/api/scaffolder/.well-known/backstage/permissions/metadata`
- `scaffolder.task.read`: prevents reading of tasks, task events/eventstreams
  - Test with the `GET /v2/tasks`, `GET /v2/tasks/:taskId`, `GET /v2/tasks/:taskId/events` and `GET /v2/tasks/:taskId/eventstream` endpoints
  - Test with the UI: View log stream, list tasks
- `scaffolder.task.create`: prevents creation of scaffolder tasks (can't trigger a software template)
  - Test with the `POST /v2/tasks` endpoint and try launching a software template via the UI (catalog entity about page, and Create page should both have their buttons disabled)
- `scaffolder.task.cancel`: prevents cancellation of scaffolder tasks
  - Test with the `POST /v2/tasks/:taskId/cancel` endpoint and try cancelling a running task in the UI (cancel button should be disabled)
